File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int size = (nums.length * (nums.length - 1) * (nums.length - 2)) / 6;
        int[] newNum = new int[size];
        int idx = 0;
        for(int i = 0; i < nums.length - 2; i++) {
            for(int j = i + 1; j < nums.length; j++) {
                for(int k = j + 1; k < nums.length; k++) {
                    newNum[idx++] = nums[i] + nums[j] + nums[k];
                }
            }
        }
    
        for(int i = 0; i < size; i++) {
            int sq = (int) Math.sqrt(newNum[i]);
            int count = 0;
            for(int j = 1 ; j <= sq; j++) {
                if(newNum[i] % j == 0) {
                    count += 2;
                }
            }
            if(sq * sq == newNum[i]) {
                count--;
            }
          
            if (count == 2) {
                answer++;
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function isPrimeNumber(num) {
    const sqrt = Math.floor(Math.sqrt(num));
    let count = 0;
    for (let i = 1; i <= sqrt; i++) {
        if (num % i === 0) count+=2;
    }
    if(sqrt * sqrt === num) count--;
  
    return count === 2 ? true : false;
}

function solution(nums) {
    let answer = 0;
 
    for(let i = 0; i < nums.length - 2; i++) {
        let sum = 0;
        for (let j = i + 1; j < nums.length - 1; j++)  {
            for (let k = j + 1; k < nums.length; k++) {
                sum = nums[i] + nums[j] + nums[k];
                 if (isPrimeNumber(sum)) {
                    answer++;
                 }
            }
        }
       
    }
    
    return answer;
}